File: go/1299_Replace_Elements_with_Greatest_Element_on_Right_Side.go
func replaceElements(arr []int) []int {
    var greatest = -1;
        
    for i := len(arr) - 1; i >= 0; i-- {
        var temp = arr[i];
        arr[i] = greatest;
        greatest = max(greatest, temp);
    }
    
    return arr;
}

func max(x, y int) int {
	if x < y {
		return y
	}
	return x
}

File: php/1200/1299_Replace_Elements_with_Greatest_Element_on_Right_Side.php
<?php

class Solution 
{

    /**
     * @param Integer[] $arr
     * @return Integer[]
     */
    function replaceElements($arr) 
    {
        $greatest = -1;
        
        for ($i = count($arr) - 1; $i >= 0; $i--) {
            $temp = $arr[$i];
            $arr[$i] = $greatest;
            $greatest = max($greatest, $temp);
        }
        
        return $arr;
    }
}

